app/vmctl/vm_native.go

```diff
@@ -303,3 +312,30 @@ func byteCountSI(b int64) string {
 	return fmt.Sprintf("%.1f %cB",
 		float64(b)/float64(div), "kMGTPE"[exp])
 }
+
+// buildMatchWithFilter adds additional filters from native.Filter.Match to discovered
+// metric names via /api/v1/series API.
+func buildMatchWithFilter(filter string, metricName string) (string, error) {
+	labels, err := promutils.NewLabelsFromString(filter)
+	if err != nil {
+		return "", err
+	}
+
+	var str strings.Builder
```

Comment:
Try replacing `strings.Builder` with a simple `[]byte`, and use `append()` instead of `WriteString()` in the code below. I bet the resulting code will become easier to read. As for `strconv.Quote()`, it can be substituted with `strconv.AppendQuote()` in this case.

Reply:
I have updated the logic. I made it simplier